lib/dialog_ai_assist/openai_client.rb

```
@@ -0,0 +1,130 @@
+      request['Authorization'] = "Bearer #{@api_key}"
+      request.body = body.to_json
+
+      begin
+        response = http.request(request)
+        handle_http_response(response)
+      rescue Net::ReadTimeout => e
+        raise Error, "Request timed out: #{e.message}"
+      rescue StandardError => e
+        raise Error, "Request failed: #{e.message}"
+      end
+    end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `make_request` is too high. [<12, 20, 2> 23.41/17]

(Metrics/AbcSize)